mirrord/layer/src/lib.rs

```diff
@@ -141,36 +141,45 @@ pub(crate) fn port_debug_patch(addr: SocketAddr) -> bool {
     }
 }
 
+/// Loads mirrord configuration and applies [`nix_devbox_patch`] patches.
+fn layer_pre_initialization() -> Result<(), LayerError> {
+    let args = std::env::args().collect::<Vec<_>>();
+    let given_process = args
+        .first()
+        .and_then(|arg| arg.split('/').last())
+        .ok_or(LayerError::NoProcessFound)?;
+
+    let mut config = std::env::var("MIRRORD_CONFIG_FILE")
+        .map(PathBuf::from)
+        .map(|path| LayerFileConfig::from_path(&path))?
+        .unwrap_or_default()
+        .generate_config()?;
+
+    nix_devbox_patch(&mut config);
+    let skip_processes = config.skip_processes.clone().map(VecOrSingle::to_vec);
+
+    if should_load(given_process, skip_processes) {
+        layer_start(config);
+    }
+
+    Ok(())
+}
+
+/// The one true start of mirrord-layer.
+#[cfg(not(test))]
```

Comment:
the only reason i used !cfg(test) was because it gave a bunch of unused code warnings, and you can also check with vscode it highlights everything yellow as unused code

Reply:
Oh I was wondering why it was like that. Guess this was fixed, as I'm not getting any unused warnings, plus vscode is highlighting everything correctly now.

It's kinda crappy because it's macro inside macro, but I'm reducing this function to be a one-liner anyway (or very close to it).

Reply:
sorry it is still there, https://github.com/metalbear-co/mirrord/actions/runs/3422857852/jobs/5700779282#step:15:153

we can either find a different solution to it or just keep using cfg!

Reply:
Back to `cfg!`!